File: packages/backend/app/schemas/product.py
from pydantic import BaseModel


class ProductSchema(BaseModel):
    id: int
    store: str
    category: str
    subcategory: str | None
    name: str
    description: str | None
    price: float | None
    unit: str | None
    product_url: str | None
    image_url: str | None
    brand: str | None
    eco_score: str | None
    in_stock: bool
    tags: list[str] | None

    class Config:
        from_attributes = True


class ProductSearchParams(BaseModel):
    query: str | None = None
    store: str | None = None
    category: str | None = None
    min_price: float | None = None
    max_price: float | None = None
    eco_score: str | None = None
    page: int = 1
    per_page: int = 20

File: packages/shared/src/types/product.ts
export type Store = 'gamma' | 'praxis' | 'ikea' | 'karwei' | 'kwantum';

export interface Product {
  id: number;
  store: Store;
  category: string;
  subcategory: string | null;
  name: string;
  description: string | null;
  price: number | null;
  unit: string | null;
  productUrl: string | null;
  imageUrl: string | null;
  brand: string | null;
  ecoScore: string | null;
  inStock: boolean;
  tags: string[];
}

export interface CartItem {
  id: number;
  projectId: number;
  product: Product | null;
  paintColor: import('./paint').PaintColor | null;
  quantity: number;
  unit: string | null;
  notes: string | null;
}

export interface ShoppingCart {
  items: CartItem[];
  totalCost: number;
  byStore: Record<Store, CartItem[]>;
}
